### ISSUE TITLE
Save file loading issue with Hard Targets Weapon modules

### ISSUE THREAD

**Issue by Godfrid** [chummer5a/chummer5a#606]
I have an issue with loading save files in case if there is any weapon with HT accessories on them (Long barrel for example). The saved character can't be loaded. Is it a unique problem or a known issue?

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.FormatException: Input string was not in a correct format.
   at System.Number.StringToNumber(String str, NumberStyles options, NumberBuffer& number, NumberFormatInfo info, Boolean parseDecimal)
   at System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
   at System.Convert.ToInt32(String value)
   at Chummer.WeaponAccessory.Load(XmlNode objNode, Boolean blnCopy) in C:\projects\chummer5a\Chummer\Classes\clsEquipment.cs:line 7859
   at Chummer.Weapon.Load(XmlNode objNode, Boolean blnCopy) in C:\projects\chummer5a\Chummer\Classes\clsEquipment.cs:line 4868
   at Chummer.Character.Load() in C:\projects\chummer5a\Chummer\Classes\clsCharacter.cs:line 1953
   at Chummer.frmMain.LoadCharacter(String strFileName, Boolean blnIncludeInMRU, String strNewName, Boolean blnClearFileName) in C:\projects\chummer5a\Chummer\frmMain.cs:line 641
   at Chummer.frmMain.OpenFile(Object sender, EventArgs e) in C:\projects\chummer5a\Chummer\frmMain.cs:line 615
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36213 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Chummer5
    Assembly Version: 5.179.0.0
    Win32 Version: 5.179.0
    CodeBase: file:///D:/RPG/Shadowrun%205/cstest1/chummer/Chummer5.exe
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36213 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36250 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36287 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36310 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36246 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36213 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------

**Comment by chummer5a** [comment]
Hrm, I can't replicate it in the dev builds; can you upload your character file somewhere for me to take a look at?

**Comment by holgerschlegel** [comment]
I can produce the same exception using the attached character created with an older nightly build (maybe 1 month old) of Chummer5.

[TextChar.zip](https://github.com/chummer5a/chummer5a/files/76600/TextChar.zip)

**Comment by chummer5a** [comment]
Can't reproduce in development or release builds. Please upload a copy of the affected .chum5 file.

**Comment by holgerschlegel** [comment]
The attached file (TextChar.zip) in my previous comment contains a character that can not be loaded with current nightly build.

**Comment by Godfrid** [comment]
Here is two save files. One is a totally new character made in the newest version, only thing it got is a weapon with a HT mod. The old one is mostly from 1.175 build: https://drive.google.com/folderview?id=0BznKy122EAN3MEkyMXF1Q3FJbzA&usp=sharing

Interesting that the 1.175 build is able to open these files without any issue.
Also as I was trying different version a new problem appeared in these versions. If I'm building a new character with priority or sum to ten, a similar bug report appears, but if I hit continue it work alright just the pop-up annoying. Now I'm presuming that the problem might be caused by the fact that I have different versions in different directories, and somehow in the background they overwrite each other? 

Another problem, if you say so I will go into detail and open another topic for it: Some metamagics aren't available for mages, aspected magicans. I fund that the problem is these metamegics hasn't been made available for these magic users in chummer, so it's I guess an easy to solve problem. For example: Flexible signatura, Flux.

**Comment by chummer5a** [comment]
Could you please test it with the newest nightly build? Sorry, we made some backend changes that broke the automatic deployment and I didn't notice til just now.

**Comment by Godfrid** [comment]
Now it looks it is good with the new character saves, but as is see there is another issue with the old one. When I try to load that I get this message:

************ Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at Chummer.Armor.Load(XmlNode objNode, Boolean blnCopy) in C:\projects\chummer5a\Chummer\Classes\clsEquipment.cs:line 1228
   at Chummer.Character.Load() in C:\projects\chummer5a\Chummer\Classes\clsCharacter.cs:line 1942
   at Chummer.frmMain.LoadCharacter(String strFileName, Boolean blnIncludeInMRU, String strNewName, Boolean blnClearFileName) in C:\projects\chummer5a\Chummer\frmMain.cs:line 641
   at Chummer.frmMain.OpenFile(Object sender, EventArgs e) in C:\projects\chummer5a\Chummer\frmMain.cs:line 615
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripButton.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36213 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Chummer5
    Assembly Version: 5.179.0.0
    Win32 Version: 5.179.0
    CodeBase: file:///D:/RPG/Shadowrun%205/cstest2/Chummer5.exe
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36213 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36250 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36287 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36310 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36246 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.36213 built by: FX452RTMLDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

**Comment by chummer5a** [comment]
Fudge, I thought that would work. Will be fixed in the next build.

**Comment by Godfrid** [comment]
Also there seems to be a problem puting the newchar into career mode, even after removed the HT upgraded stuffs:

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at Chummer.frmCreate.SaveCharacterAsCreated() in C:\projects\chummer5a\Chummer\Character Creation\frmCreate.cs:line 18859
   at Chummer.frmCreate.SaveCharacter() in C:\projects\chummer5a\Chummer\Character Creation\frmCreate.cs:line 18761
   at Chummer.frmCreate.tsbSave_Click(Object sender, EventArgs e) in C:\projects\chummer5a\Chummer\Character Creation\frmCreate.cs:line 2319
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripButton.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by chummer5a** [comment]
It's something I broke with the most recently used character files, I
think. It can't find the career mode file to save it. Currently looking
into it.

**Comment by Godfrid** [comment]
Despite the error message, after restarting chummer, and loading the file, the character have been put into career mode.

**Comment by chummer5a** [comment]
Yup, should have clarified that. It can't find the character to load, but
the conversion's already been done.Should be able to get it fixed for the
next build.

**Comment by Godfrid** [comment]
I made a few test. It seems the build works with 1.175 saves, which haven't been saved with the previous build. I tried HT upgrades, and it works with these characters on the current build, only the last version affected saves are buggy, and there is the put in to career mode error message.
Again thanks for your work and sorry for the lot of comment, just trying to be as specific with the problems as I can to help locate the source of the issue.
